### PR TITLE
[ONNX] Support aten::hstack and aten::vstack

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102872
* #103061

https://github.com/microsoft/onnxscript/pull/762 is actually not used by FX graph.
Fixes https://github.com/microsoft/onnx-converters-private/issues/168

### CODE REVIEW

File: torch/onnx/symbolic_opset11.py [hunk 1]
```diff
@@ -1558,3 +1560,45 @@ def prim_constant_chunk(g: jit_utils.GraphContext, self, chunks, dim):
...
+    )
+    utils._add_output_to_block(else_context_equal.block, result_else)
+    result = if_op_greater.node().output()
+
+    return result
+
+
+@_onnx_symbolic("aten::vstack")
+@_beartype.beartype
+def vstack(g: jit_utils.GraphContext, tensor_list: _C.Value):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:1601]: For symbolic functions, an old pattern is to look if the tensor list is a static list, i.e., prim::ListConstruct. And if so, individual checks and validation similar to `at_least2d` can be applied. This usually results in a more efficient graph, but we could also argue it applies constant folding/optimization prematurely.

Taking a step back, it seems logical to support `aten::at_least2d`, and invoke it here. I feel this is also the best solution for torchlib, plus post optimizations. For that we optionally need OnnxFunction calling OnnxFunction in torchlib, and need ability to write loop over Sequence in OnnxFunction.

Comment (titaiwangms) [reply torch/onnx/symbolic_opset11.py:1601]: I thought about unpack the list to for loop the tensors. But another difficulty is that a IF condition has to be set for each tensor to see if they are at least 2D or not (basically the whole at_least2D). If this part of logic is also written in constant, Then 2D promotion would be added into graph regardless of next inputs are 2D already.

Comment (BowenBao) [reply torch/onnx/symbolic_opset11.py:1601]: yea, usually it is accompanied with the assumption that input ranks remain the same. Hence more suited for post optimization.

Comment (titaiwangms) [reply torch/onnx/symbolic_opset11.py:1601]: OK I understand it now. It's kind of like the idea of op.Transpose that we usually flip them the same way in all batches.

Comment (titaiwangms) [reply torch/onnx/symbolic_opset11.py:1709]: Done

**Comment by titaiwangms** [comment]
@BowenBao This one is also good to be merged. The hstack and vstack is decomposed to ues aten::cat in FX graph. https://[github.com/microsoft/onnxscript/pull/762](https://github.com/microsoft/onnxscript/pull/762) is blocked by a subtle bug in ORT. But in converter side, it's all good to go.

**Review by justinchuby — APPROVED** [review]

**Pull request closed without merging**